### PR TITLE
NIFI-9751 - Poll as needed during system-tests to ensure expected ser…

### PR DESCRIPTION
**Pull request by greyp9**
…ver state

#### Description of PR
- Additional polling as needed to ensure expected server state.
- Some additional logging to help identify process shutdown failures.
- Adjust certain timings to reduce noise in logs during troubleshooting. 
- Remove test condition that needs further investigatation (NIFI-9717).
- System test log messages should include timestamps to help correlate to corresponding NiFi logs.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-system-tests/nifi-system-test-suite/src/test/resources/simplelogger.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+org.slf4j.simpleLogger.showDateTime=true
```
Comment (greyp9) [inline nifi-system-tests/nifi-system-test-suite/src/test/resources/simplelogger.properties:1]: Added this in order to generate timestamps for system-test-suite log messages.  The timestamps are very useful for correlating test case trace to the corresponding NiFi trace.

File: nifi-system-tests/nifi-system-test-suite/src/test/resources/conf/clustered/node2/logback.xml [hunk 2]
```diff
@@ -84,13 +84,13 @@
...
         <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
-            <pattern>%date %level [%thread] %logger{40} %msg%n</pattern>
+            <pattern>%date %level [%thread] %logger %msg%n</pattern>
         </encoder>
     </appender>
 
     <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
         <encoder class="ch.qos.logback.classic.encoder.PatternLayoutEncoder">
-            <pattern>%date %level [%thread] %logger{40} %msg%n</pattern>
+            <pattern>%date %level [%thread] %logger %msg%n</pattern>
```
Comment (greyp9) [inline nifi-system-tests/nifi-system-test-suite/src/test/resources/conf/clustered/node2/logback.xml:93]: The `{40}` makes it harder to grep log files for log messages coming from a given package.

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/loadbalance/LoadBalanceIT.java [hunk 3]
```diff
@@ -283,32 +287,11 @@ private long getQueueBytes(final String connectionId) {
...
-        }
-
-        return false;
-    }
-
     private boolean isConnectionDoneLoadBalancing(final String connectionId) {
         try {
             final ConnectionEntity connectionEntity = getNifiClient().getConnectionClient().getConnection(connectionId);
             final String loadBalanceStatus = connectionEntity.getComponent().getLoadBalanceStatus();
+            logger.trace("LoadBalanceStatus = [{}]", loadBalanceStatus);
```
Comment (greyp9) [inline nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/loadbalance/LoadBalanceIT.java:294]: This helped identify a few cases where the load balance subsystem failed.

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/clustering/FlowSynchronizationIT.java [hunk 4]
```diff
@@ -822,8 +847,9 @@ public void testUnnecessaryProcessorsAndConnectionsRemoved() throws NiFiClientEx
 
         final ProcessorEntity terminate = getClientUtil().createProcessor("TerminateFlowFile");
         final ConnectionEntity connection = getClientUtil().createConnection(generate, terminate, "success");
-        getClientUtil().startProcessor(terminate);
         getClientUtil().startProcessor(generate);
+        waitForMinQueueCount(connection.getId(), 1);
```
Comment (greyp9) [inline nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/clustering/FlowSynchronizationIT.java:851]: This was a weird case where the `waitForQueueCount(0)` query happened before the generate processor fired its event.

File: nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/NiFiClientUtil.java [hunk 5]
```diff
@@ -1256,7 +1256,7 @@ public ProvenanceEntity waitForComplete(final ProvenanceEntity entity) throws In
         verificationRequest.setRequest(requestDto);
 
         VerifyConfigRequestEntity results = nifiClient.getReportingTasksClient().submitConfigVerificationRequest(verificationRequest);
-        while (!results.getRequest().isComplete()) {
+        while ((!results.getRequest().isComplete()) || (results.getRequest().getResults() == null)) {
```
Comment (greyp9) [inline nifi-system-tests/nifi-system-test-suite/src/test/java/org/apache/nifi/tests/system/NiFiClientUtil.java:1259]: This covers a transitory state where the server reports `COMPLETED`, but before it had assembled the results.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making these improvements @greyp9. The results look good from local testing and should improve the behavior of GitHub automated builds. +1 merging.

**Pull request closed without merging**